pkg/proxy/iptables/proxier.go

```diff
@@ -734,12 +748,18 @@ func (proxier *Proxier) syncProxyRules() {
 		klog.V(4).Infof("syncProxyRules took %v", time.Since(start))
 	}()
 
+	// don't sync rules till we've received services, endpoints and node's
+	// metadata if TopologyAwareServiceRouting is enabled.
+	if !proxier.isAllSynced() {
+		klog.V(2).Info("Not syncing iptables rules until Services, Endpoints and Node's Metadata (if TopologyAwareServiceRouting is enabled) have been received from master")
```

Comment:
nit: can't we tell here if the feature gate is enabled and adjust the message accordingly?

Reply:
Nice catch! I'll make a change for this.